datalake/src/main/java/com/sequenceiq/datalake/cm/ClouderaManagerRangerUtil.java

```diff
@@ -118,17 +109,61 @@ public boolean isCloudIdMappingSupported(String stackCrn) throws ApiException {
         return isCloudIdMappingSupported(rolesResourceApi, clusterName, rangerUserSyncRoleName);
     }
 
-    public Optional<ApiCommand> setAzureCloudIdentityMapping(String stackCrn, Map<String, String> azureUserMapping) throws ApiException {
+    private Map<String, String> userMappingStrToMap(String userMappingStr) {
+        userMappingStr = userMappingStr.trim();
+        if (userMappingStr.isEmpty()) {
+            return Collections.emptyMap();
+        }
+        Map<String, String> ret = new HashMap<>();
+        String[] mappings = userMappingStr.split(";");
+        Arrays.stream(mappings).forEach(mapping -> {
+            String[] entry = mapping.split("=");
+            String key = entry[0];
+            String val = entry[1];
+            ret.put(key, val);
+        });
+        return ret;
+    }
+
+    private Map<String, String> getExistingAzureUserMapping(RolesResourceApi rolesResourceApi, String clusterName, String rangerUserSyncRoleName)
+            throws ApiException {
+        ApiConfigList roleConfigList = rolesResourceApi.readRoleConfig(clusterName, rangerUserSyncRoleName, RANGER_SERVICE_NAME, "summary");
+        ApiConfig azureUserMappingConfig = roleConfigList.getItems().stream()
+                .filter(apiConfig -> apiConfig.getName().equals(AZURE_USER_MAPPING))
+                .findFirst()
+                .get();
```

Comment:
what if none found? are we sure there is at least one?

Reply:
I am not sure if we can be sure that the config is always present--probably not, but in any case we should protect against this possibility. Now checks that the optional value is present before getting it. If it is absent, an empty map is returned.